server/src/test/java/org/elasticsearch/health/metadata/HealthMetadataSerializationTests.java

```diff
@@ -58,7 +58,11 @@ protected ClusterState.Custom mutateInstance(ClusterState.Custom instance) {
     }
 
     private static HealthMetadata randomHealthMetadata() {
-        return new HealthMetadata(randomDiskMetadata());
+        return new HealthMetadata(randomDiskMetadata(), randomShardLimitsMetadata());
+    }
+
+    private static HealthMetadata.ShardLimits randomShardLimitsMetadata() {
+        return new HealthMetadata.ShardLimits(randomInt(), randomInt());
```

Comment:
I am not 100% sure but I think `randomInt()` generates negative values too. So, could you double check this to avoid adding a flaky test when the validation for only positive numbers is added.